src/System.Text.Json/src/System/Text/Json/ThrowHelper.Serialization.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.CompilerServices;
+using System.Text.Json.Serialization;
+
+namespace System.Text.Json
+{
+    internal static partial class ThrowHelper
+    {
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        public static void ThrowArgumentException_DeserializeWrongType(Type type, object value)
+        {
+            throw new ArgumentException(SR.Format(SR.DeserializeWrongType, type.FullName, value.GetType().FullName));
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        public static void ThrowJsonReaderException_DeserializeUnableToConvertValue(Type propertyType, in Utf8JsonReader reader, in ReadStack state)
+        {
+            throw GetJsonReaderException_DeserializeUnableToConvertValue(propertyType, reader, state);
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        public static JsonReaderException GetJsonReaderException_DeserializeUnableToConvertValue(Type propertyType, in Utf8JsonReader reader, in ReadStack state)
```

Comment:
Yes I can refactor this a bit now that I removed all of the extensibility code. Previously there was a call to the this where I wanted it public\internal to keep the `throw` in the calling code to prevent a dummy line after throwhelper call to initialize an `out` parameter.